doc/fr/developper/writestreamconnector.rst

```diff
@@ -0,0 +1,586 @@
+
+.. image:: /_static/images/developper/lua/add_parameter.png
+   :align: center
+
+Then click **Save** and go to generate the configuration and restart **cbd**.
+
+.. note::
+   Don’t forget to restart “centengine” too to create the Centreon Broker cache.
+
+Data still stored into **/var/log/centreon-broker/bbdo2file.log** log file::
+
+  name => error
+  category => 3
+  interval => 300
+  rrd_len => 3456000
+  value => 0
+  value_type => 0
+  type => 196612
+  ctime => 1522315660
+  index_id => 4880
+  element => 4
+  state => 0
+  category => 3
+  interval => 300
+  rrd_len => 3456000
+  is_for_rebuild => false
+  service_id => 1056
+  type => 196609
+  ctime => 1522315660
+  host_id => 145
+  element => 1
+  is_for_rebuild => false
+  metric_id => 11920
+
+Manipulate data
+***************
+
+Not all data are necessary. We will select only the NEB category and the events 
+regarding hosts and services status to write into a file the following data:
+
+* The type of object (HOST or SERVICE)
+* The name of host and ID
+* The description of service and ID (for service only)
+* The status and output of service
+
+To do this, we need to select not all events received by Centreon Broker but only 
+event regarding status of hosts and services objects from NEB category. By 
+following `the technical documentation <https://documentation.centreon.com/docs/centreon-broker/en/latest/dev/bbdo.html#event-categories>`_, 
+the NEB events type (real time) are linked to category 1. Moreover, the status of 
+host is link to element 14 and 24 for the status of services. To filter on these 
+events only, we will add a filter function with these properties.
+
+The new LUA script will contain:
+
+.. code-block:: lua
+
+  function init(conf)
+    logFile = conf['logFile']
+    broker_log:set_parameters(3, "/var/log/centreon-broker/debug.log")
+  end
+  
+  function write(d)
+    local output = ""
+  
+    local host_name = broker_cache:get_hostname(d.host_id)
+    if not host_name then
+      broker_log:info(3, "Unable to get name of host, please restart centengine")
+      host_name = d.host_id
+    end
+  
+    if d.element == 14 then
+      output = "HOST:" .. host_name .. ";" .. d.host_id .. ";" .. d.state .. ";" .. d.output
+      wrintIntoFile(output)
+      broker_log:info(output)
+    elseif d.element == 24 then
+      local service_description = broker_cache:get_service_description(d.host_id, d.service_id)
+      if not service_description then
+        broker_log:info(3, "Unable to get description of service, please restart centengine")
+        service_description = d.service_id
+      end
+      output = "SERVICE:" .. host_name .. ";" .. d.host_id .. ";" .. service_description .. ";" .. d.service_id .. ";" .. d.state .. ";" .. d.output
+      wrintIntoFile(output)
+      broker_log:info(output)
+    end
+    return true
+  end
+  
+  function wrintIntoFile(output)
+    local file,err = io.open(logFile, 'a')
```

Comment:
Have you notice that sometimes, you writes strings as "titi", and sometimes as 'titi' ?
In Lua, there are no performance differences between them. The only real difference is that if the string is _l'hiver_, you can write "l'hiver" and if the string is _une "bête"_, you can write 'une "bête"'.
And if you always want to use "", you can too with "une \"bête\"".
If you don't explain, it is strange to see the changes between each lines.